src/pam_hbac_config.c

```diff
@@ -281,13 +281,20 @@ ph_read_config(pam_handle_t *pamh,
 
     logger(pamh, LOG_DEBUG, "config file: %s", config_file);
 
+#ifdef HPUX
+    errno = 666;       // for some reason when invoked through pam in HPUX, fopen doesn't set errno on error
```

Comment:
Do you have any idea why errno is not set when in PAM context? When invoked otherwise it sets it properly. For instance when using the `pam_test_client` utility the correct error is set.